frontends/p4-14/typecheck.cpp

```diff
@@ -26,6 +26,16 @@ limitations under the License.
 // - set type for Member and HeaderStackItemRefs
 class TypeCheck::AssignInitialTypes : public Transform {
     const IR::V1Program   *global = nullptr;
+
+    template <typename NodeType, typename TypeType>
+    void setType(NodeType* currentNode, const TypeType* type) {
+        BUG_CHECK(currentNode == getCurrentNode<NodeType>(),
+                  "Expected to be called on the visitor's current node");
+        currentNode->type = type;
+        if (type != getOriginal<NodeType>()->type)
+            LOG1("Set initial type " << type << " for expression " << currentNode);
```

Comment:
This should probably be LOG2 or even LOG3

( see https://github.com/p4lang/p4c/blob/master/docs/README.md -- towards the end)